src/fsharp/NameResolution.fs

```diff
@@ -2496,6 +2496,15 @@ let ChooseTyconRefInExpr (ncenv: NameResolver, m, ad, nenv, id: Ident, typeNameR
 /// that may represent further actions, e.g. further lookups.
 let rec ResolveExprLongIdentPrim sink (ncenv: NameResolver) first fullyQualified m ad nenv (typeNameResInfo: TypeNameResolutionInfo) (id: Ident) (rest: Ident list) isOpenDecl =
     let resInfo = ResolutionInfo.Empty
+    let canSuggestThisItem (item:Item) =
+        // All items can be suggested except nameof when it comes from FSharp.Core.dll and the nameof feature is not enabled
+        match item with
+        | Item.Value v ->
+            let isNameOfOperator = valRefEq ncenv.g ncenv.g.nameof_vref v
+            if isNameOfOperator && not (ncenv.g.langVersion.SupportsFeature LanguageFeature.NameOf) then false
```

Comment:
Can we refractor this so that the language support test is done only once per compiler run? I.e move it up to the top, remove the parameter from the outer function and return a new function that takes the item

Reply:
@forki, it can't actually be global, because it's a per project setting, and in the IDE we will have compilations with different settings for it.  So  I added it to the NameResolution context.  Which seems to be a decent compromise.